From 40799b33b55fec7c39bc1d30785667b915d8e0bf Mon Sep 17 00:00:00 2001
From: Yelena Krivosheev <yelena@marvell.com>
Date: Mon, 12 Nov 2018 13:30:10 +0200
Subject: [PATCH 0789/1051] net: mvneta: fix size of buf_virt_addr

The array of RX buffers pointers is allocated only once on init.
Size of this array must be max RX queue size.

Change-Id: I08d8651dba901dee725f2ada4ec030a07276bc77
Signed-off-by: Yelena Krivosheev <yelena@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60855
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvneta.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvneta.c b/drivers/net/ethernet/marvell/mvneta.c
index ddc934dc65bd..9158a375af3b 100644
--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -4450,7 +4450,7 @@ static int mvneta_init(struct device *dev, struct mvneta_port *pp)
 		rxq->time_coal = MVNETA_RX_COAL_USEC;
 		rxq->buf_virt_addr
 			= devm_kmalloc_array(pp->dev->dev.parent,
-					     rxq->size,
+					     MVNETA_MAX_RXD,
 					     sizeof(*rxq->buf_virt_addr),
 					     GFP_KERNEL);
 		if (!rxq->buf_virt_addr)
-- 
2.17.1

